### ISSUE TITLE
HashLocationStrategy routing from a nested child router-outlet to a top-level route fails

### ISSUE THREAD

**Issue by btgoodwin** [angular/angular#12564]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
[My SO question](http://stackoverflow.com/questions/40285582/angular2-routing-from-child-routes-to-root-routes-with-hashlocationstrategy)

From within a nested router-outlet, a component tries to use `router.navigate` (or `navigateByUrl` or `routerLink` on a DOM element) to navigate to some top-level route (example at the link, TopB).  The router seems to reject that valid name and instead fail over to the top-level redirect path no matter what I try (preceding route with a /, etc.).

If I drop the `useHash: true` from my AppModule (i.e., drop HashLocationStrategy) and define my APP_BASE_HREF as `/apps/App1` (i.e., the static route of index.html per my Tornado server), the routing behavior works as expected.  The downside is I can no longer "reload" at my browser from within the router hierarchy since the URL is then treated as a static file by the Tornado server (which of course it cannot find).

Granted, this "reload" limitation might be overcome by some level of tornado configuration settings, but that's not the point of this issue.  The issue is that by using HashLocationStrategy, nested child components cannot navigate to valid top-level routes.  I've verified that the instance of `router` at the nested child has the top-level `config` and can even find the named path I'm trying to traverse.

**Expected behavior**
A nested child component should be able to route to any defined route regardless of LocationStrategy.

**Minimal reproduction of the problem with instructions**
Please see the SO post linked above; it contains instructions.

**What is the motivation / use case for changing the behavior?**
Right now, enabling HashLocationStrategy seems to break the ability of the Router to navigate to parent routes even though the instance of Router injected to the child is indeed the top-level router with configuration.  Disabling HashLocationStrategy allows these paths to be traversed.

**Please tell us about your environment:**
CentOS 6.7 64-bit (in a VM, running the web server)
    Tornado 4.4.2

macOS 10.12.1 (running the browser)

* **Angular version:** 2.0.X
    Angular 2.1.0
    Router 3.1.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Chrome 53.0
Safari 10.0
No others tested
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript 2.0.3
ES5

* **Node (for AoT issues):** `node --version` =

**Comment by DzmitryShylovich** [comment]
Pls provide a minimal working reproduction in plnkr.

**Comment by btgoodwin** [comment]
[No?](https://plnkr.co/edit/Ujnxbo13DN4QkrZWKbHO?p=preview)

In all seriousness: this can be closed/deleted.

I apologize for wasting anyone's time.  The plnkr I patched together in the above link works with both methods I accused of not working, and HashLocationStrategy.  So I went digging into my code and I think over some iteration and lack of coffee, the `<a (click)='function()'></a>` method I managed to insert an `href="#"` that I kept overlooking.  So naturally, clicking that link would kick me back to the default route.

I've deleted the SO post above since this came down to a typo in my code.  I'll leave the plnkr though since it's a reasonable example showing nested child routes, etc.

My apologies again.
